compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5587,12 +5593,32 @@ private STNode parseStatementStartsWithIdentifier(SyntaxKind nextTokenKind, STNo
             case SEMICOLON_TOKEN:
                 // Here we directly start parsing as a statement that starts with an expression.
                 return parseStamentStartWithExpr(nextTokenKind, identifier);
+            case QUESTION_MARK_TOKEN:
+                // if the next token is question-mark then it is an optional type descriptor with user defined type
+                switchContext(ParserRuleContext.VAR_DECL_STMT);
+                STNode questionMarkToken = parseQuestionMark();
+                STNode optionalTypeDesc = STNodeFactory.createOptionalTypeDescriptorNode(identifier, questionMarkToken);
+                //check whether if there is more type descriptor elements
+                STNode typeDesc = parseComplexTypeDescriptor(optionalTypeDesc);
+                varName = parseVariableName();
+                finalKeyword = STNodeFactory.createEmptyNode();
+                return parseVarDeclRhs(annots, finalKeyword, typeDesc, varName, false);
```

Comment:
These three lines seems to be repeating in three places inside this method. Better to take it out as a method and re-use it.

Reply:
done